### PR TITLE
Texture Loading

### LINKED ISSUE #21: [10] Add support for textures
As a user, I'd like to visualize the textures in my model.
- Load textures using ASSIMP
- Allocate textures in GPU memory
- Create samplers for textures
- Update default shader to handle textures

### PR DESCRIPTION
**Pull request by tigerman9854**
Added the ability to load a single texture.  I think this is good enough because we don't want to complicate our shader with a ton of texture abilities.

This completes half of #21

### CODE REVIEW

File: ModelViewer/ModelViewer/ModelLoader.cpp [hunk 1]
```diff
@@ -72,8 +73,32 @@ Mesh ModelLoader::ProcessMesh(aiScene const* pScene, uint meshIdx)
...
-	// TODO: Handle textures
-
+	// Search for an ambient, diffuse, or specular texture.  We will only load the first one we find because our shader
+	// only handles a single texture
+	std::vector<aiTextureType> validTextureTypes = { aiTextureType_AMBIENT, aiTextureType_DIFFUSE, aiTextureType_SPECULAR };
+	newMesh.m_hasTexture = false;
+	for (auto it : validTextureTypes)
+	{
+		// Only load 1 texture
+		if (newMesh.m_hasTexture) {
```
Comment (richjoregonstate) [inline ModelViewer/ModelViewer/ModelLoader.cpp:83]: Do we want to prompt the user to select a valid texture? Might be a 'sprint 4' kind of issue XD

**Review by richjoregonstate — APPROVED** [review]
Looks good :)

File: ModelViewer/ModelViewer/ModelLoader.cpp (hunk 1 above)
Comment (tigerman9854) [reply ModelViewer/ModelViewer/ModelLoader.cpp:83]: I'm not really a fan of user prompts (slows down our quick loading).  This method is very likely to load the correct type of texture for what our shader is capable of.  The other texture types are like emissive, normals, and bumpmapped stuff.

Comment (tigerman9854) [reply ModelViewer/ModelViewer/ModelLoader.cpp:83]: In the future, we could support all that, but it would be a major pain :)

**Pull request merged**